test/metabase/models/collection/graph_test.clj

```diff
@@ -434,3 +434,12 @@
     (with-n-temp-users-with-personal-collections 2000
       (is (>= (db/count Collection :personal_owner_id [:not= nil]) 2000))
       (is (map? (graph/graph))))))
+
+(deftest modify-perms-for-app-collections-test
+  (testing "that we cannot modify perms for app collections"
+    (mt/with-temp* [Collection [{coll-id :id}]
+                    App [_app {:collection_id coll-id}]]
+      (is (thrown? clojure.lang.ExceptionInfo
```

Comment:
it would be better to use `thrown-with-msg?` here to check that you're actually getting the error message you expect (`Cannot set app permissions using this endpoint`) as opposed to this endpoint failing for some other reason and not actually testing what you think you're testing